src/main/java/org/reaktivity/nukleus/http2/internal/stream/Http2ServerFactory.java

```diff
@@ -2807,19 +2883,20 @@ private void onRequestWindow(
                 if (requestBudgetId != 0L && requestDebitorIndex == NO_DEBITOR_INDEX)
                 {
                     requestDebitor = supplyDebitor.apply(budgetId);
-                    requestDebitorIndex = requestDebitor.acquire(budgetId, initialId, this::flushRequestDataIfNecessary);
+                    requestDebitorIndex = requestDebitor.acquire(budgetId, initialId, decodeNetworkIfNecessary);
                 }
 
-                flushRequestDataIfNecessary(traceId);
+                decodeNetworkIfNecessary(traceId);
 
-                if (requestSlot == NO_SLOT)
+                if (dataLength == dataBytesConsumed)
```

Comment:
Now that we are no longer buffering at the application, how can the `END` be triggered too early?

Reply:
what if we don't have enough credit to flush all data